sdk/core/core/src/az_span.c

```diff
@@ -15,20 +17,42 @@ enum
   AZ_ASCII_LOWER_DIF = 'a' - 'A',
 };
 
-AZ_NODISCARD az_result
-az_span_slice(az_span span, int32_t low_index, int32_t high_index, az_span* out_sub_span)
+void az_precondition_failed_default()
 {
-  // left part
-  az_span left = span;
-  if (high_index >= 0)
+  /* By default, when a precondition fails the calling thread is suspended forever */
+  while (true)
   {
-    left = az_span_take(span, high_index);
+    az_platform_sleep_msec(INT32_MAX);
```

Comment:
This hangs the program forever in case a test fails. The gate will be stalled on a test and not print the result of a test should it fail.

Reply:
yeah, that's currently something that made us remove all tests validating fail cases...

Reply:
okay can this be sure to wait at least to Friday to merge because I do not understand this change and
> Remove all tests validating fail cases

sounds really odd to me

Reply:
totally.
I think @JeffreyRichter  wanted to have this PR to better show how things would look using PRECONDITION instead of CONTRACT.

I should have mentioned this in the PR description (sorry about that).

Let's ask @barcharcraz to add some minutes in the Friday meeting agenda for this.

Reply:
Awesome thanks Victor

Reply:
@danewalton : For background on this change...
Yuxin & I had a long conversation about contract/preconditions/argument checking and together, we arrived at this precondition solution. This past Monday, I met with Eric, John, and Himanshu to discuss the change and its impact. They all agreed to this change. So, feel free to speak to them to learn more, I'm also happy to talk to you about it and we can wait until Friday to merge..